source/common/router/router.cc

```diff
@@ -630,6 +605,64 @@ void Filter::onUpstreamReset(UpstreamResetType type,
   }
 }
 
+bool Filter::maybeRetryReset(const Http::StreamResetReason reset_reason) {
+  Upstream::HostDescriptionConstSharedPtr upstream_host;
+  if (upstream_request_) {
```

Comment:
Mind moving this inside the if-block below? We only care about the upstream host if there is a retry state